python/gfs_geostrophic.py

```diff
@@ -56,9 +58,14 @@
 
 
 # open dataset and retreive variables
-url = 'https://thredds.ucar.edu/thredds/dodsC/grib/NCEP/GFS/' + \
-    'Global_0p25deg_ana/GFS_Global_0p25deg_ana_' + mydate + '_' + \
-    myhour + '00.grib2'
+# url = 'https://thredds.ucar.edu/thredds/dodsC/grib/NCEP/GFS/' + \
+#     'Global_0p25deg_ana/GFS_Global_0p25deg_ana_' + mydate + '_' + \
+#     myhour + '00.grib2'
+
+url = (
+    "https://thredds.ucar.edu/thredds/dodsC/grib/NCEP/GFS/"
+    f"Global_0p25deg_ana/GFS_Global_0p25deg_ana_{sample_datetime:%Y%m%d_%H}00.grib2"
+)
 
```

Comment:
Herein lies the crux of using the `datetime` object as an object: using formatted strings, you can get away with things like this.